spec/lib/saml/post_url_service_spec.rb

```diff
@@ -603,52 +603,105 @@
+                  let(:review_instance_slug) { 'some-review-instance-slug' }
+                  let(:review_instance_url) { "#{review_instance_slug}.review.vetsgov-internal" }
 
-                before { allow(Settings).to receive(:review_instance_slug).and_return(review_instance_slug) }
+                  before { allow(Settings).to receive(:review_instance_slug).and_return(review_instance_slug) }
 
-                it 'has a login redirect url as a parameter embedded in review instance terms of use page' do
-                  expect(subject.terms_of_use_redirect_url)
-                    .to eq("http://#{review_instance_url}/terms-of-use?#{expected_redirect_url_param}")
+                  it 'has a login redirect url as a parameter embedded in review instance terms of use page' do
+                    expect(subject.terms_of_use_redirect_url)
+                      .to eq("http://#{review_instance_url}/terms-of-use?#{expected_redirect_url_param}")
+                  end
+                end
+
+                context 'and authentication is not occurring on a review instance' do
+                  it 'has a login redirect url as a parameter embedded in terms of use page with success' do
+                    expect(subject.terms_of_use_redirect_url)
+                      .to eq("#{values[:base_redirect]}/terms-of-use?#{expected_redirect_url_param}")
+                  end
                 end
               end
 
-              context 'and authentication is not occurring on a review instance' do
+              context 'when tracker application is nil' do
+                let(:application) { nil }
+
                 it 'has a login redirect url as a parameter embedded in terms of use page with success' do
                   expect(subject.terms_of_use_redirect_url)
                     .to eq("#{values[:base_redirect]}/terms-of-use?#{expected_redirect_url_param}")
                 end
               end
-            end
 
-            context 'when tracker application is nil' do
-              let(:application) { nil }
+              context 'when tracker application is not within TERMS_OF_USE_ENABLED_CLIENTS' do
+                let(:application) { 'some-application' }
 
-              it 'has a login redirect url as a parameter embedded in terms of use page with success' do
-                expect(subject.terms_of_use_redirect_url)
-                  .to eq("#{values[:base_redirect]}/terms-of-use?#{expected_redirect_url_param}")
+                it 'has a login redirect url with success not embedded in a terms of use page' do
+                  expect(subject.terms_of_use_redirect_url)
+                    .to eq(expected_login_redirect_url)
+                end
               end
             end
 
-            context 'when tracker application is not within TERMS_OF_USE_ENABLED_CLIENTS' do
-              let(:application) { 'some-application' }
+            context 'with unverified user' do
```

Comment:
I don't think any of the unverified users should be seeing the ToU page, something is going on with these tests.

My guess would be that the user created for these tests is associated with a `user_account` that has an `icn` (which would mean the user was verified at some point). So I would make sure this isn't happening with these tests

Reply:
OK, just changed the user set-up to explicitly set the `icn` to nil:

```ruby
let(:user) { build(:user, :loa1, icn: nil) }
```

However, I ran the specs again and they all passed, when they should have failed based on what you are saying (unless I misinterpreted). Here is the data from the `user`:

<img width="1025" alt="Screenshot 2023-10-24 at 1 09 09 PM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/10838930/5b6433dd-6f62-4b79-906b-5ec78cd14bbe">

See anything amiss here? If not, I wonder if there is a bug after all.

Reply:
`needs_accepted_terms_of_use` is `false`, which should mean ToU is not being shown. So the user model appears to be correct at least

Reply:
Can you hit a breakpoint in sessions controller where this happens?
`if Settings.vsp_environment != 'production' && @current_user.needs_accepted_terms_of_use`

Reply:
errr, or maybe what I'd do is combine the 'unverified' and 'verified' contexts since it doesn't matter which is which

Reply:
Yeah, I essentially copied the specs for the verified user flow for use in the unverified user flow, modifying user set-up for each context. So, really, the spec I made was redundant and unnecessary. Dropping this commit; thanks for bringing my attention to it.